extensions/guacamole-auth-quickconnect/src/main/resources/controllers/quickconnectController.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The controller for making ad-hoc (quick) connections
+ */
+angular.module('guacQuickConnect').controller('quickconnectController', ['$scope', '$injector', '$log',
+        function manageConnectionController($scope, $injector, $log) {
+
+    // Required types
+    var ClientIdentifier    = $injector.get('ClientIdentifier');
+    var Connection          = $injector.get('Connection');
+
+    // Required services
+    var $location                = $injector.get('$location');
+    var $routeParams             = $injector.get('$routeParams');
+    var guacNotification         = $injector.get('guacNotification');
+    var connectionService        = $injector.get('connectionService');
+    var schemaService            = $injector.get('schemaService');
+
+    /**
+     * An action to be provided along with the object sent to showStatus which
+     * closes the currently-shown status dialog.
+     */
+    var ACKNOWLEDGE_ACTION = {
+        name        : "MANAGE_CONNECTION.ACTION_ACKNOWLEDGE",
+        // Handle action
+        callback    : function acknowledgeCallback() {
+            guacNotification.showStatus(false);
+        }
+    };
+
+    $scope.uri = null;
+
+    $scope.selectedDataSource = 'quickconnect';
+
+    /**
+     * Saves the connection, creating a new connection or updating the existing
+     * connection.
+     */
+    $scope.quickConnect = function quickConnect() {
+
+        // Construct parameters from URI...
+        /**
+         * Parse URL into the following components:
+         * [0] - Full URL
+         * [3] - Protocol
+         * [5] - Username
+         * [7] - Password
+         * [8] - Hostname
+         * [10] - Port
+         * [11] - Path
+         * [13] - Document
+         * [15] - Parameters
+         * [17] - JS Route
+         */
+        var regexURL = /^(((rdp|ssh|telnet|vnc)?):\/)?\/?((.*?)(:(.*?)|)@)?([^:\/\s]+)(:([^\/]*))?((\/\w+\/)*)([-\w.\/]+[^#?\s]*)?(\?([^#]*))?(#(.*))?$/g;
```

Comment:
Whoa.

Reply:
Ha.  Yeah.  If you have suggestions for better ways to parse out a URI, I'm open.  That regex right there is the result of a combo of Google searches and working with the regex on regex101.com.

Reply:
Obviously it would be better if the available protocols there were parsed from a REST API check of supported protocols instead of hard-coded.

Reply:
On a similar note, any reason why you chose to handle this client-side vs. server-side? I'm not sure whether one is better than the other at the moment, but I am curious. Might be worth considering.

Reply:
I think the reason I did it that way was because I was trying to leverage as much as possible of the existing REST endpoints for creating/saving the connections, which seems to be easier if the connection attributes and parameters are parsed out on the client-side and then fed to the existing API.

I suppose I could leverage the extension-specific REST API and feed the entire connection string to a REST endpoint and then parse it out there.  Or is there some other way I'm not thinking about that already exists to get the data to the server-side for parsing by the extension?

Reply:
> I'm not sure whether one is better than the other at the moment, but I am curious.

Well, since Java already has a URI class and is capable of parsing out the URI without the...colorful...regex, I think I might be leaning toward redoing it server-side.  JavaScript does have some basic parsing available through the DOM, and some external libraries available for it, but it doesn't seem nearly as robust as the Java library.